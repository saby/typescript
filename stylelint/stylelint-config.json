{
  "customSyntax": "postcss-less",
  "rules": {
    "at-rule-no-unknown": true,
    "block-no-empty": true,
    "color-no-invalid-hex": true,
    "comment-no-empty": true,
    "declaration-block-no-duplicate-properties": [
      true,
      {
        "ignore": [
          "consecutive-duplicates-with-different-values"
        ]
      }
    ],
    "declaration-block-no-shorthand-property-overrides": true,
    "font-family-no-duplicate-names": true,
    "function-linear-gradient-no-nonstandard-direction": true,
    "keyframe-declaration-no-important": true,
    "no-descending-specificity": true,
    "no-duplicate-at-import-rules": true,
    "no-duplicate-selectors": true,
    "no-empty-source": true,
    "no-invalid-double-slash-comments": true,
    "property-no-unknown": [
      true,
      {
        "ignoreProperties": [
          "container-name",
          "container-type"
        ]
      }
    ],
    "selector-pseudo-class-no-unknown": true,
    "selector-pseudo-element-no-unknown": true,
    "selector-type-no-unknown": true,
    "string-no-newline": true,
    "at-rule-empty-line-before": [
      "always",
      {
        "except": [
          "blockless-after-same-name-blockless",
          "first-nested"
        ],
        "ignore": [
          "after-comment"
        ]
      }
    ],
    "comment-empty-line-before": [
      "always",
      {
        "except": [
          "first-nested"
        ],
        "ignore": [
          "stylelint-commands"
        ]
      }
    ],
    "comment-whitespace-inside": "always",
    "custom-property-empty-line-before": [
      "always",
      {
        "except": [
          "after-custom-property",
          "first-nested"
        ],
        "ignore": [
          "after-comment",
          "inside-single-line-block"
        ]
      }
    ],
    "function-name-case": "lower",
    "length-zero-no-unit": [true, {
      "ignore": ["custom-properties"]
    }],
    "rule-empty-line-before": [
      "always-multi-line",
      {
        "except": [
          "first-nested"
        ],
        "ignore": [
          "after-comment"
        ]
      }
    ],
    "selector-pseudo-element-colon-notation": "double",
    "value-keyword-case": [
      "lower",
      {
        "ignoreFunctions": ["extend"]
      }
    ],
    "font-family-no-missing-generic-family-keyword": [
      true,
      {
        "ignoreFontFamilies": "/^cbuc-icons\\d*$/"
      }
    ],
    "color-named": "never",
    "color-hex-length": "short",
    "shorthand-property-no-redundant-values": true,
    "declaration-block-single-line-max-declarations": 0,
    "selector-max-id": 0,
    "selector-max-type": 0,
    "selector-type-case": "lower",
    "selector-max-universal": 0,
    "no-unknown-animations": true,
    "declaration-no-important": true,
    "unit-no-unknown": true,
    "declaration-empty-line-before": "never"
  }
}
