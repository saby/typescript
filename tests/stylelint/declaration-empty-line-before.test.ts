import { lint } from 'stylelint';
import {checker} from './checker';
const config = require('../../stylelint/stylelint-config.json');

describe('declaration-empty-line-before', () => {
    describe('ошибки', () => {
        const cases = [
            `.class {
   height: 1px;

   color: #ccc;
}
`,
            `.class {

   height: 1px;
}
`
        ];

        cases.forEach((code, index) => {
            it(`индекс теста: ${index}`, async () => {
                const result = await lint({
                    code,
                    config
                });

                checker(result);
            });
        });
    });

    describe('не ошибки', () => {
        const cases = [
            `.class {
   height: 1px;
   color: #ccc;
}
`,
            `.class {
   /* comment */
   height: 1px;
   color: #ccc;
}
`, `.class {
   height: 1px;

   /* comment */
   color: #ccc;
}
`,
            `.class { height: 1px; }
`
        ];

        cases.forEach((code, index) => {
            it(`индекс теста: ${index}`, async () => {
                const result = await lint({
                    code,
                    config
                });

                checker(result);
            });
        });
    });
});
