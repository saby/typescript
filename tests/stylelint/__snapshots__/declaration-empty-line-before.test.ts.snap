// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`declaration-empty-line-before не ошибки индекс теста: 0 1`] = `Array []`;

exports[`declaration-empty-line-before не ошибки индекс теста: 0 2`] = `Array []`;

exports[`declaration-empty-line-before не ошибки индекс теста: 1 1`] = `Array []`;

exports[`declaration-empty-line-before не ошибки индекс теста: 1 2`] = `Array []`;

exports[`declaration-empty-line-before не ошибки индекс теста: 2 1`] = `Array []`;

exports[`declaration-empty-line-before не ошибки индекс теста: 2 2`] = `Array []`;

exports[`declaration-empty-line-before не ошибки индекс теста: 3 1`] = `
Array [
  Object {
    "column": 8,
    "endColumn": 24,
    "endLine": 1,
    "line": 1,
    "rule": "declaration-block-single-line-max-declarations",
    "severity": "error",
    "text": "Expected no more than 0 declarations (declaration-block-single-line-max-declarations)",
  },
]
`;

exports[`declaration-empty-line-before не ошибки индекс теста: 3 2`] = `Array []`;

exports[`declaration-empty-line-before ошибки индекс теста: 0 1`] = `
Array [
  Object {
    "column": 4,
    "endColumn": 16,
    "endLine": 4,
    "line": 4,
    "rule": "declaration-empty-line-before",
    "severity": "error",
    "text": "Unexpected empty line before declaration (declaration-empty-line-before)",
  },
]
`;

exports[`declaration-empty-line-before ошибки индекс теста: 0 2`] = `Array []`;

exports[`declaration-empty-line-before ошибки индекс теста: 1 1`] = `
Array [
  Object {
    "column": 4,
    "endColumn": 16,
    "endLine": 3,
    "line": 3,
    "rule": "declaration-empty-line-before",
    "severity": "error",
    "text": "Unexpected empty line before declaration (declaration-empty-line-before)",
  },
]
`;

exports[`declaration-empty-line-before ошибки индекс теста: 1 2`] = `Array []`;
